w3-blob.md

```diff
@@ -0,0 +1,199 @@
+} representation keyed
+
+type BlobAddAllocation struct {
+  content   Multihash
+  size      Int
+
+  url       URL
+  headers   HTTPHeaders
+}
+
+type UCANLocationClaim struct {
+  iss     DID
+  aud     DID
+  sub     SpaceDID
+  cmd     "assert/location"
+  args    LocationClaim
+  nonce   bytes
+  exp     Int
+}
+
+type LocationClaim struct {
+  content    Multihash
+  url        URL
+  range      ByteRange
+}
+
+type ByteRange struct {
+  offset    Int
+  length    Int
+} representation tuple
+```
+
+#### Add Blob Success
+
+##### Add Blob Allocated
+
+Capability provider MUST issue receipt with `BlobAddAllocation` success result if underlying space has enough capacity for the requested blob, but has not such blob locally.
+
+The `url` field MUST be set to the HTTP PUT endpoint where `content` matching specified [multihash] and `size` could be uploaded.
+
+The `headers` field MUST be set to the HTTP headers to be send along with `content` to the HTTP PUT endpoint.
+
+###### Add Blob Committed
+
+Capability provider MUST issue receipt with `LocationClaim` capability delegated to the [space] when requested blob is added to the space, which can happen if provider already has a matching blob locally.
+
+By delegating `LocationClaim` to the space provider makes a commitment to serve corresponding blob at byte range in from given URL to all authorized agents.
+
+> Not that since `aud` of the claim is the `space` DID, space can authorize other principals by re-delegating this claim.
+
+## Get Blob
+
+Authorized agent MAY invoke `/space/content/get/blob` capability on the [space] subject to query a state of the corresponding blob.
+
+### Get Blob Capability
+
+#### Get Blob Capability Schema
+
+```ipldsch
+type struct GetBlob {
+  cmd   "/space/content/get/blob"
+  sub   SpaceDID
+  args  BlobQuery
+}
+
+type struct BlobQuery {
+  content   Multihash
+}
+```
+
+### Get Blob Receipt
+
+#### Get Blob Result
+
+```ipldsch
+type BlobAddResult union {
+  BlobGetSuccess    "ok"
+  BlobGetFailure    "error"
+} representation keyed
+
+type BlobGetSuccess union {
+  BlobAddAllocation     "allocated"
+  &UCANLocationClaim    "committed"
+} representation keyed
+
+type BlobGetFailure struct {
+  message     string
+}
+```
+
+#### Get Blob Success
+
+Capability provider MUST return the state of the requested blob. If memory for the blob was allocated but content has not been yet uploaded result MUST be of `BlobAddAllocation` case. If blob has been allocated and uploaded result MUST of `UCANLocationClaim` case.
+
+> Please note that implementation MAY check local state on invocation and update from `BlobAddAllocation` to `UCANLocationClaim`, however this SHOULD be done in referentially transparent way. In other words no side effects should be exhibited, instead function should lazily compute and cache.
+
+## Publish Blob
+
+Blob can be published by authorizing read interface (e.g. IPFS gateway) by re-delegating `LocationClaim` for a corresponding blob to it.
```

Comment:
I think we need more details here. This is supposed to be for the location of the multihash of the blob, or every single block within the blob (which would mainly happen in IPNI use case)

Reply:
No it is the whole blob, equivalent of accessing car from gateway by CAR cid, except in this case it would be raw CID. Block level index can be then accomplished through inclusions claims that effectively say that this multihash is a slice of that other multihash at the given byte offset and this number of bytes

Reply:
What if the Blob gets moved around by w3s? Does it need to be redelegated?

Reply:
No it does not per https://github.com/web3-storage/RFC/pull/13 we intend to use locations that are agnostic of actual blob site, they can HTTP redirect as necessary. Also more intelligent clients could recognize hints in the URL to go directly to a site without going through a redirect. The best thing is we always have a fallback and ability to changes things without having to distribute updates.